### PR TITLE
Prepare for 1.23.0 release

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,107 @@
...
+* Add resource assertion methods to SpanDataAssert and MetricAssert
+  ([#5160](https://github.com/open-telemetry/opentelemetry-java/pull/5160))
+
+
+### Semantic Conventions
+
+* Update the semantic conventions to v1.17.0
+  ([#5134](https://github.com/open-telemetry/opentelemetry-java/pull/5134))
+* Update semconv to 1.18.0
+  ([#5188](https://github.com/open-telemetry/opentelemetry-java/pull/5188))
```
Comment (trask) [inline CHANGELOG.md:90]:
```suggestion
* Update semconv to 1.18.0
  ([#5188](https://github.com/open-telemetry/opentelemetry-java/pull/5188))
```

**Review by trask — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jack-berg) [reply CHANGELOG.md]: I'll merge these into one entry, but include links to both PRs so folks can quickly find the details of each update.

**Review by jkwatson — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,106 @@
...
+  If using configuring OTLP programatically,
+  use `Otlp*MetricExporterBuilder#setDefaultAggregationSelector(DefaultAggregationSelector)`.
+* The metrics SDK undergone significant internal refactoring which results in reduced complexity and
+  significantly reduced memory allocation during metric collection / export.
+
+### API
+
+#### Logs
+
+* WARNING: Split out Event API from Log API. Event API is now published in `opentelemetry-api-logs`.
```
Comment (mateuszrzeszutek) [inline CHANGELOG.md:21]: Shouldn't that be `opentelemetry-api-events`?
```suggestion
* WARNING: Split out Event API from Log API. Event API is now published in `opentelemetry-api-events`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,106 @@
...
+  API / SDK usage has also changed - see PR for details.
+  ([#5049](https://github.com/open-telemetry/opentelemetry-java/pull/5049))
+
+### SDK
+
+* Add shutdown / close to OpenTelemetrySdk
```
Comment (mateuszrzeszutek) [inline CHANGELOG.md:27]: Minor nit: do we want to use code-quotes for all class/methods names? Some of them are formatted, some of them note
```suggestion
* Add shutdown / close to `OpenTelemetrySdk`
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (jack-berg) [reply CHANGELOG.md]: Good point. Added formatting to all the places it was missing.

File: CHANGELOG.md (hunk 2 above)
Comment (jack-berg) [reply CHANGELOG.md]: Good catch

**Pull request merged**